components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectSystemClaimImplTest.java

```diff
@@ -0,0 +1,144 @@
+        oAuthTokenReqMessageContext = new OAuthTokenReqMessageContext(oAuth2AccessTokenReqDTO);
+        oAuth2AccessTokenRespDTO = new OAuth2AccessTokenRespDTO();
+        oAuth2AuthorizeReqDTO = new OAuth2AuthorizeReqDTO();
+        oAuthAuthzReqMessageContext = new OAuthAuthzReqMessageContext(oAuth2AuthorizeReqDTO);
+        oAuth2AuthorizeRespDTO = new OAuth2AuthorizeRespDTO();
+        openIDConnectSystemClaim = new OpenIDConnectSystemClaimImpl();
+    }
+
+    @DataProvider(name = "getAuthzAdditionalClaims")
+    public Object[][] getAuthzAdditionalClaims() throws Exception {
+
+        return new Object[][] {
+                {"code", AUTHORIZATION_CODE, ACCESS_TOKEN, getHashValue(AUTHORIZATION_CODE),
+                        getHashValue(ACCESS_TOKEN)},
+                {"token", AUTHORIZATION_CODE, ACCESS_TOKEN, EMPTY_VALUE, getHashValue(ACCESS_TOKEN)},
+                {"id_token", AUTHORIZATION_CODE, ACCESS_TOKEN, EMPTY_VALUE, EMPTY_VALUE}
+        };
+    }
+
+    @Test(dataProvider = "getAuthzAdditionalClaims")
+    public void testAuthzGetAdditionalClaims(String responseType,
+                                             String authorizationCode, String accessToken,
+                                             String authorizationHashCode, String hashAccessToken) throws Exception {
+
+        oAuth2AuthorizeRespDTO.setAuthorizationCode(authorizationCode);
+        oAuth2AuthorizeReqDTO.setResponseType(responseType);
+        oAuth2AuthorizeRespDTO.setAccessToken(accessToken);
+        Map<String, Object> claims = openIDConnectSystemClaim.getAdditionalClaims(oAuthAuthzReqMessageContext,
+                oAuth2AuthorizeRespDTO);
+        Assert.assertEquals(claims.get("at_hash"), hashAccessToken);
+        Assert.assertEquals(claims.get("c_hash"), authorizationHashCode);
+    }
+
+    @DataProvider(name = "getAdditionalClaims")
+    public Object[][] getAdditionalClaims() throws Exception {
+
+        return new Object[][] {
+                {AUTHORIZATION_CODE, ACCESS_TOKEN, getHashValue(AUTHORIZATION_CODE), getHashValue(ACCESS_TOKEN)},
+                {EMPTY_VALUE, "accessToken", EMPTY_VALUE, getHashValue("accessToken")}
+
+        };
+    }
+
+    @Test(dataProvider = "getAdditionalClaims")
+    public void testGetAdditionalClaims(String authorizationCode, String accessToken,
+                                        String authorizationHashCode, String hashAccessToken) throws Exception {
+
+        oAuth2AccessTokenReqDTO.setAuthorizationCode(authorizationCode);
+        oAuth2AccessTokenRespDTO.setAccessToken(accessToken);
+        Map<String, Object> claims = openIDConnectSystemClaim.getAdditionalClaims(oAuthTokenReqMessageContext,
+                oAuth2AccessTokenRespDTO);
+        Assert.assertEquals(claims.get("at_hash"), hashAccessToken);
+        Assert.assertEquals(claims.get("c_hash"), authorizationHashCode);
+    }
+    
+    private String getHashValue(String value) throws IdentityOAuth2Exception {
```

Comment:
If there are any private methods, you can use WhiboxImp and invoke that method.  `Whitebox.invokeMethod(openIDConnectSystemClaim, "getHashValue", value`);